src/compiler/crystal/compiler.cr

```diff
@@ -376,15 +377,15 @@ module Crystal
           cmd = "#{cl} #{Process.quote_windows("@" + args_filename)}"
         end
 
-        {cmd, nil}
+        {cl, cmd, nil}
       elsif program.has_flag? "wasm32"
         link_flags = @link_flags || ""
-        { %(wasm-ld "${@}" -o #{Process.quote_posix(output_filename)} #{link_flags} -lc #{program.lib_flags}), object_names }
+        {"wasm-ld", %(wasm-ld "${@}" -o #{Process.quote_posix(output_filename)} #{link_flags} -lc #{program.lib_flags}), object_names}
       else
         link_flags = @link_flags || ""
         link_flags += " -rdynamic"
 
-        { %(#{CC} "${@}" -o #{Process.quote_posix(output_filename)} #{link_flags} #{program.lib_flags}), object_names }
+        {CC, %(#{CC} "${@}" -o #{Process.quote_posix(output_filename)} #{link_flags} #{program.lib_flags}), object_names}
```

Comment:
The actual executable could be different from `CC` if the environment variable is something like `FOO=BAR gcc`. But I don't have a good solution for that because `parse_arguments` is not sufficient for arbitrary shell commands

Reply:
Yeah, I think that's good enough. And definitely an improvement over printing the full linker command line which can easily span multiple pages.

I suppose if we *had* a good solution for that, we also wouldn't need to run `Process.run(shell: true)` through `/bin/sh` 🤷